source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -0,0 +1,795 @@
+                            "1",
+                            Patterns::Integer(0),
+                            "The number of subdivisions in the East-West direction.");
+          prm.declare_entry("North-South subdivision",
+                            "1",
+                            Patterns::Integer(0),
+                            "The number of subdivisions in the North-South direction.");
+          prm.declare_entry("Depth subdivision",
+                            "1",
+                            Patterns::Integer(0),
+                            "The number of subdivisions in depth.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    //initialize several static member variables
+    template <int dim> double EllipsoidalChunk<dim>::semi_major_axis_a = 1;
+    template <int dim> double EllipsoidalChunk<dim>::Eccentricity = 1;
+    template <int dim> double EllipsoidalChunk<dim>::semi_minor_axis_b = 1;
+    template <int dim>  double EllipsoidalChunk<dim>::rot_para_to_para_angle = 0;
+    template <int dim>  double EllipsoidalChunk<dim>::para_to_rect_angle = 0;
+    template <int dim>  double EllipsoidalChunk<dim>::rotation_latitude = 0;
+    template <int dim>  double EllipsoidalChunk<dim>::rotation_longitude = 0;
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Ellipsoidal chunk");
+        {
+          /**
+           * Get latitude and longitudes defining region of interest from
+           * the parameter file.
+           */
+          corners.resize(4);
+          std::string NEcorner = prm.get("NE corner");
+          std::string NWcorner = prm.get("NW corner");
+          std::string SWcorner = prm.get("SW corner");
+          std::string SEcorner = prm.get("SE corner");
+
+          /**
+           * make a list of what corners are present and check that there should be one or two corner missing,
+           * otherwise throw an exception
+           */
+          std::vector<bool> present(4,true);
+          unsigned int missing = 0;
+          if (NEcorner == "")
+            {
+              missing++;
+              present[0] = false;
+            }
+          if (NWcorner == "")
+            {
+              missing++;
+              present[1] = false;
+            }
+          if (SWcorner == "")
+            {
+              missing++;
+              present[2] = false;
+            }
+          if (SEcorner == "")
+            {
+              missing++;
+              present[3] = false;
+            }
+
+          AssertThrow (missing != 0,
+                       ExcMessage ("Only two or three of the four corners points should be provided."));
+          AssertThrow (missing == 1 || missing == 2,
+                       ExcMessage ("Please provide two or three corners points."));

```

Comment:
I suppose this geometry model is not supposed to be run in 2d, right? What is going to happen if one tries? Can we gracefully error out early already, say at the top of this function (the first time the member functions of this class will be called after the dimension is known)?